google-http-client/src/main/java/com/google/api/client/http/HttpResponseException.java

```diff
@@ -185,7 +212,19 @@ public Builder(HttpResponse response) {
       // message
       StringBuilder builder = computeMessageBuffer(response);
       if (content != null) {
-        builder.append(StringUtils.LINE_SEPARATOR).append(content);
+        try {
+          // charset
+          charset = response.getContentCharset();
+          String strContent = getContent();
+          if (strContent != null) {
+            builder.append(StringUtils.LINE_SEPARATOR).append(strContent);
+          }
+        } catch (IOException exception) {
+          // it would be bad to throw an exception while throwing an exception
```

Comment:
We should do a log statement rather than just a print stack trace, if there are other examples of logging.

Reply:
@sduskis, there is no examples and this library is not logging.

Reply:
Done